libs/utils/trace.py

```diff
@@ -674,6 +677,23 @@ def _sanitize_CpuFrequency(self):
             return
         df = self._dfg_trace_event('cpu_frequency')
         clusters = self.platform['clusters']
+
+        # devlib introduces fake cpu_frequency events to make sure the
+        # frequency is known at the "beginning" of a trace
+        if self.hasEvents('cpu_frequency_devlib'):
+            devlib_freq = self._dfg_trace_event('cpu_frequency_devlib')
+            devlib_freq.rename(columns={'cpu_id':'cpu'}, inplace=True)
+            devlib_freq.rename(columns={'state':'frequency'}, inplace=True)
+            # Only use fake events if introduced before any cpufreq governor
+            # did, otherwise they may end up being interleaved with real events
+            # and make the the frequency coherency check fail
+            if len(devlib_freq) > 0:
+                first_devlib_freq = devlib_freq.iloc[:self.platform['cpus_count']]
+                if df.index[0] > first_devlib_freq.index[-1]:
+                    df = pd.concat([first_devlib_freq, df])
+            setattr(self.ftrace.cpu_frequency, 'data_frame', df)
```

Comment:
Fair enough.  I agree on not moving stuff that needs platform information that is only available in LISA.  It's just that we have some very basic sanitizing in trappy (e.g., in `cpu_capacity` we change the column names, https://github.com/ARM-software/trappy/blob/master/trappy/sched.py#L83 ) and it looked like some of the sanitization done in LISA is similar to that.